substrate-archive/src/database.rs

```diff
@@ -272,3 +275,111 @@ impl Insert for Metadata {
 		.map_err(Into::into)
 	}
 }
+
+#[async_trait]
+impl Insert for Traces {
+	async fn insert(mut self, conn: &mut DbConn) -> DbReturn {
+		log::debug!("Inserting Trace Data");
+		let mut batch = Batch::new(
+			"state_tracing",
+			r#"
+			INSERT INTO "state_traces" (
+				block_num, hash, is_event, timestamp, duration, file, line, trace_id, trace_parent_id, target, name, traces
+			) VALUES
+			"#,
+			r#"
+			ON CONFLICT DO NOTHING
+			"#,
+		);
+
+		let block_num = self.block_num();
+		let hash = self.hash().to_vec();
```

Comment:
The hash is cloned here because `self.spans` and `self.events` are moved?

I guess the hash small so probably fine.

Reply:
actually was able to make it all references